### PR TITLE
[project-base] webline layout

### PR DESCRIPTION
**Pull request by TomasGottvald**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Remove global webline and container. Splits web content to smaller web-lines with own container. Brings better comfort while building new modern design.
|New feature| Yes 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig [hunk 1]
```diff
@@ -12,162 +12,164 @@
...
             <div class="in-action">
-                {% if cart.itemsCount > 0 %}
-                    <div class="in-action__right">
-                        {{ form_row(form.submit, { label: 'Order [verb]'|trans, attr: { class: 'btn btn--success in-action__btn in-action__btn--big'}}) }}
-                    </div>
-                {% endif %}
-
                 <div class="in-action__left">
-                    <a href="{{ url('front_homepage') }}" class="btn in-action__btn">
+                    <a href="{{url('front_homepage') }}" class="btn in-action__btn">
```
Comment (topolanekmartin) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig]: Should not be white space before `url`?

File: docs/frontend/page-layout-design.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Page layout design
+Page layout is divided to three main parts
+- web__header
+- web__main
+- web__footer
+
+## Weblines and containers
+Each part described above should contain `web__line` - this is full width div which wraps every line.
+Inside every `web__line` it must be `web__container` - it has set `web-width` and makes web content centered
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply docs/frontend/page-layout-design.md]: Yes in some edge cases it is not important to have container there.

File: docs/frontend/page-layout-design.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+in div with `web-width` width.
+
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply docs/frontend/page-layout-design.md]: fixed

File: docs/frontend/page-layout-design.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+### Weblines
+`web__main` is devided to rows. We call this rows as `web__lines`.  If you use design without left panel you can see on homepage these weblines:
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply docs/frontend/page-layout-design.md]: yes, thx

File: docs/frontend/index.md [hunk 5]
```diff
@@ -1,7 +1,12 @@
 # Frontend
 
+## Getting started
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply docs/frontend/index.md]: fixed

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less [hunk 6]
```diff
@@ -35,35 +35,51 @@ img {
...
+
+        &--no-padding {
+            padding: 0;
+        }
+
+        .web__main--with-panel & {
+            padding: 0;
+            width: 100%;
+
+            @media @query-xl {
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: This is intention - bcs. this definitions are for layout with panel.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less [hunk 7]
```diff
@@ -35,35 +35,51 @@ img {
...
+                width: calc(~"@{web-width} - @{web-panel-width} - 2*@{web-padding}");
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: I deleted whole part, width: 100% is sufficient. Thx

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig [hunk 8]
```diff
@@ -12,162 +12,164 @@
...
+                            {% set cartItemDiscount = cartItemDiscounts[index] %}
+                            <tr class="table-cart__row js-cart-item">
+                                <td class="table-cart__cell table-cart__cell--image">
+                                    <a href="{{ url('front_product_detail', {id: cartItem.product.id}) }}" title="{{ cartItem.product.name }}" class="table-cart__cell__image">
+                                        {{ image(cartItem.product, { size: 'thumbnail' }) }}
+                                    </a>
+                                </td>
+
+                                <td class="table-cart__cell table-cart__cell--name js-cart-item-name">
+                                    <a href="{{ url('front_product_detail', {id: cartItem.product.id}) }}" title="">
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig]: I don´t think so... we should focus on these part during accesability issues. Deleted

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less [hunk 9]
```diff
@@ -35,35 +35,51 @@ img {
...
         &--with-panel {
+            padding: 0 @web-padding-small;
+
             @media @query-lg {
                 display: flex;
                 width: 100%;
             }
+
+            @media @query-vl {
+                padding: 0 @web-padding;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: Usualy web padding is set by container. But in layout with panel are containers on right side content without padding (line 58) and padding is set by web__main.

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig (hunk 1 above)
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig]: Yes, fixed, thx.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less [hunk 10]
```diff
@@ -1,5 +1,5 @@
-@web-footer-with-cookies-margin-bottom: 36px;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: removed

File: docs/frontend/page-layout-design.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply docs/frontend/page-layout-design.md]: I think so, thanks for the update.

File: docs/frontend/page-layout-design.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply docs/frontend/page-layout-design.md]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less (hunk 10 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less (hunk 6 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: Ah, thanks for the clarification.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less (hunk 7 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: Thanks

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig (hunk 1 above)
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig (hunk 8 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig]: Okay, I will satisfy with this.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less (hunk 9 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less]: Alright.

**Comment by TomasGottvald** [comment]
I added better html format in twig blocks

File: docs/frontend/page-layout-design.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
+# Page layout design
+Page layout is divided to three main parts
+- web__header
```
Comment (DavidOstrozlik) [inline docs/frontend/page-layout-design.md]: hmm if this is talking about classes, shouldn't it be with a dot? `.web__header` (and the other classes as well)?

Comment (TomasGottvald) [reply docs/frontend/page-layout-design.md]: Thx for your note. Yes it will be better to use dot before class names. Fixed

Comment (TomasGottvald) [reply docs/frontend/page-layout-design.md]: I added updgrade notes for this changes.

**Pull request merged**